alicloud/data_source_alicloud_cs_kubernetes_permissions.go

```diff
@@ -0,0 +1,110 @@
+package alicloud
+
+import (
+	cs "github.com/alibabacloud-go/cs-20151215/v2/client"
+	openapi "github.com/alibabacloud-go/darabonba-openapi/client"
+	"github.com/alibabacloud-go/tea/tea"
+	"github.com/aliyun/terraform-provider-alicloud/alicloud/connectivity"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/hashcode"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func dataSourceAlicloudCSKubernetesPermissions() *schema.Resource {
+	return &schema.Resource{
+		Read: dataAlicloudCSKubernetesPermissionsRead,
+
+		Schema: map[string]*schema.Schema{
+			"uid": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"permissions": {
+				Optional: true,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"resource_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"resource_type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"cluster", "namespace", "console"}, false),
+						},
+						"role_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"role_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"is_owner": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntInSlice([]int{0, 1}),
+						},
+						"is_ram_role": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntInSlice([]int{0, 1}),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataAlicloudCSKubernetesPermissionsRead(d *schema.ResourceData, meta interface{}) error {
+	client, err := cs.NewClient(&openapi.Config{
```

Comment:
Please move the client to connectivity.

Reply:
update in :https://github.com/aliyun/terraform-provider-alicloud/pull/3495/commits/cf231d5079bb6e37dae54dbec0743935e6fee91c